server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -677,10 +677,16 @@ public enum FieldExtractPreference {
          * Load the field from doc-values into a BlockLoader supporting doc-values.
          */
         DOC_VALUES,
+        /** Loads the field by extracting the extent from the binary encoded representation */
+        EXTRACT_SPATIAL_BOUNDS,
         /**
          * No preference. Leave the choice of where to load the field from up to the FieldType.
          */
-        NONE
+        NONE;
+
+        public static FieldExtractPreference forColumnReader(boolean columnReader) {
+            return columnReader ? DOC_VALUES : NONE;
```

Comment:
This does not really make sense to me. Both DOC_VALUES and EXTRACT_SPATIAL_BOUNDS will require a column reader. I suspect the situation is that the places this is used actually never test the EXTRACT_SPATIAL_BOUNDS case, so this does not matter. But once we add tests for that, it will.

Reply:
Yes, this is for backward compatibility. I'll just inline it so it's clearer.